src/main/clm_varcon.F90

```diff
@@ -154,6 +154,15 @@ module clm_varcon
   real(r8), public :: c14ratio = 1.e-12_r8
   ! real(r8) :: c14ratio = 1._r8  ! debug lets set to 1 to try to avoid numerical errors
 
+  !------------------------------------------------------------------
+  ! Surface roughness constants
+  !------------------------------------------------------------------
+  real(r8), public, parameter :: beta_param = 7.2_r8  ! Meier et al. (2022) https://doi.org/10.5194/gmd-15-2365-2022
+  real(r8), public, parameter :: nu_param = 1.5e-5_r8  ! Meier et al. (2022) kinematic viscosity of air
+  real(r8), public, parameter :: b1_param = 1.4_r8  ! Meier et al. (2022) empirical constant
+  real(r8), public, parameter :: b4_param = -0.31_r8  ! Meier et al. (2022) empirical constant
+  real(r8), public, parameter :: cd1_param = 7.5_r8  ! Meier et al. (2022) originally from Raupach (1994)
```

Comment:
By reverting the code to magic numbers rather than using these parameters in their place, I have matched the baseline (dev129) for the last failing test.

Reply:
I now hypothesize that the last failing test had roundoff diffs due to a change in order of operations when using `b4_param` from -0.31 to + (-0.31). Not tested.

Reply:
When cheyenne's compute nodes come up:
Try the same test, but reverting only b4_param to its hardwired number rather than all the above parameters.

Reply:
Repeated the test as recommended in the previous post:
```
+++ b/src/biogeophys/FrictionVelocityMod.F90
@@ -579,9 +579,9 @@ contains
           if (frac_sno(c) > 0._r8) then
              if(use_z0m_snowmelt) then
                 if ( snomelt_accum(c) < 1.e-5_r8 )then
-                    z0mg(c) = exp(-b1_param * rpi * 0.5_r8 + b4_param) * 1.e-3_r8
+                    z0mg(c) = exp(-b1_param * rpi * 0.5_r8 -0.31_r8  ) * 1.e-3_r8
                 else
-                    z0mg(c) = exp(b1_param * (atan((log10(snomelt_accum(c))+0.23_r8)/0.08_r8)) + b4_param) * 1.e-3_r8
+                    z0mg(c) = exp(b1_param * (atan((log10(snomelt_accum(c))+0.23_r8)/0.08_r8)) -0.31_r8  ) * 1.e-3_r8
                 end if
              else
                 z0mg(c) = this%zsno
@@ -594,9 +594,9 @@ contains
           if (frac_sno(c) > 0._r8) then ! Do snow first because ice could be snow-covered
              if(use_z0m_snowmelt) then
                 if ( snomelt_accum(c) < 1.e-5_r8 )then
-                    z0mg(c) = exp(-b1_param * rpi * 0.5_r8 + b4_param) * 1.e-3_r8
+                    z0mg(c) = exp(-b1_param * rpi * 0.5_r8 -0.31_r8  ) * 1.e-3_r8
                 else
-                    z0mg(c) = exp(b1_param * (atan((log10(snomelt_accum(c))+0.23_r8)/0.08_r8)) + b4_param) * 1.e-3_r8
+                    z0mg(c) = exp(b1_param * (atan((log10(snomelt_accum(c))+0.23_r8)/0.08_r8)) -0.31_r8  ) * 1.e-3_r8
```
```
+++ b/src/biogeophys/LakeFluxesMod.F90
@@ -351,9 +351,9 @@ contains
          else                          ! use roughness over snow as in Biogeophysics1
             if(use_z0m_snowmelt) then
                if ( snomelt_accum(c) < 1.e-5_r8 ) then
-                  z0mg(p) = exp(-b1_param * rpi * 0.5_r8 + b4_param) * 1.e-3_r8
+                  z0mg(p) = exp(-b1_param * rpi * 0.5_r8 -0.31_r8  ) * 1.e-3_r8
                else
-                  z0mg(p) = exp(b1_param * (atan((log10(snomelt_accum(c))+0.23_r8)/0.08_r8)) + b4_param) * 1.e-3_r8
+                  z0mg(p) = exp(b1_param * (atan((log10(snomelt_accum(c))+0.23_r8)/0.08_r8)) -0.31_r8  ) * 1.e-3_r8
                end if
             else
                z0mg(p) = params_inst%zsno
@@ -604,9 +604,9 @@ contains
             else ! Snow layers
                if(use_z0m_snowmelt) then
                   if ( snomelt_accum(c) < 1.e-5_r8 )then
-                      z0mg(p) = exp(-b1_param * rpi * 0.5_r8 + b4_param) * 1.e-3_r8 
+                      z0mg(p) = exp(-b1_param * rpi * 0.5_r8 -0.31_r8  ) * 1.e-3_r8 
                   else
-                      z0mg(p) = exp(b1_param * (atan((log10(snomelt_accum(c))+0.23_r8)/0.08_r8)) + b4_param) * 1.e-3_r8
+                      z0mg(p) = exp(b1_param * (atan((log10(snomelt_accum(c))+0.23_r8)/0.08_r8)) -0.31_r8  ) * 1.e-3_r8
```
and had a successful result (showing the FAILs only):
```
20230804_093711_tp00hq: 1 test
    FAIL ERS_Ly5_P144x1.f10_f10_mg37.IHistClm51BgcCrop.cheyenne_intel.clm-cropMonthOutput NLCOMP
    FAIL ERS_Ly5_P144x1.f10_f10_mg37.IHistClm51BgcCrop.cheyenne_intel.clm-cropMonthOutput BASELINE /glade/p/cgd/tss/ctsm_baselines/ctsm5.1.dev131: FIELDLIST field lists differ (otherwise bit-for-bit)
```

Reply:
This has been verified so resolving.